File: substring.py
T = input()
ans_list = []
for tc in range(T):
	N = input()
	S = raw_input()
	ans = 0
	for i in range(N):
		if S[i] == '0':
			continue
		for j in range(i,N):
			if S[j] == '1':
				ans = ans + 1
	ans_list.append(ans)
for a in ans_list:
    print(a)

File: substring.c
#import<stdio.h>
void main()
{
  int T,N,i,j,a,tc,ans[100000];
  char S[100000];
  scanf("%d",&T);
  for(tc=0 ; tc < T ; tc++)
    {
      a = 0;
      scanf("%d",&N);
      scanf("%s",S);
      for(i = 0; i < N ; i++)
	{
	  if(S[i] == '0')
	    continue;
	  for(j = i ; j < N ; j++)
	    {
	      if(S[j] == '1')
		a++;
	    }
	}
      ans[tc] = a;
    }
  for(i = 0;i < T;i++)
    printf("%d\n",ans[i]);
  //return 0;
}
